src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -337,6 +343,47 @@ end
   return nothing
 end
 
+# TODO: nonconservative terms; remove type restriction on the volume_flux
+#       used for dispatch for development
+@inline function split_form_kernel!(du::AbstractArray{<:Any,4}, u,
+                                    element, mesh::TreeMesh{2},
+                                    nonconservative_terms::Val{true}, equations,
+                                    volume_flux::Tuple{Any,Any}, dg::DGSEM, cache, alpha=true)
+  # true * [some floating point value] == [exactly the same floating point value]
+  # This can (hopefully) be optimized away due to constant propagation.
+  @unpack derivative_split = dg.basis
+  symmetric_flux, nonconservative_flux = volume_flux
+
+  # Apply the symmetric flux as usual
+  split_form_kernel!(du, u, element, mesh, Val(false), equations, symmetric_flux, dg, cache, alpha)
+
+  # Calculate the remaining volume terms using the nonsymmetric generalized flux
+  for j in eachnode(dg), i in eachnode(dg)
+    u_node = get_node_vars(u, equations, dg, i, j, element)
+
+    # The diagonal terms are zero since the diagonal of `derivative_split`
+    # is zero. We ignore this for now.
+
+    # x direction
+    integral_contribution = zero(u_node)
+    for ii in eachnode(dg)
+      u_node_ii = get_node_vars(u, equations, dg, ii, j, element)
+      flux1 = nonconservative_flux(u_node, u_node_ii, 1, equations)
```

Comment:
Maybe for consistency (or to avoid confusion) we could name it `noncons_flux1` instead of `flux1`and so on.

Reply:
Good point. I did that in c6964e8d.